src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/QPSolver.hpp

```diff
@@ -0,0 +1,139 @@
+         * @param convexHullMatrix The convex hull matrix related to allowable next step position;
+         * @return true/false in case of success/failure.
+         */
+        bool setConstraintsMatrix(const iDynTree::Vector2& currentDcmPosition, const iDynTree::Vector2& currentZmpPosition,
+                                  const iDynTree::MatrixDynSize& convexHullMatrix);
+
+        /**
+         * Set or update the gradient
+         * @param zmpWeight weight of next step position term in cost function.
+         * @param dcmOffsetWeight weight of dcm offset term in cost function.
+         * @param sigmaWeight weight of next step timing term in cost function.
+         * @param zmpNominal vector of nominal values of next step position .
+         * @param dcmOffsetNominal vector of nominal values of dcm offset.
+         * @param sigmaNominal vector of nominal values of exp(w*steptiming) .
+         * @return true/false in case of success/failure.
+         */
+        bool setGradientVector(const iDynTree::Vector2& zmpWeight, const iDynTree::Vector2& dcmOffsetWeight,
+                               const double& sigmaWeight,const iDynTree::Vector2& zmpNominal,
+                               const iDynTree::Vector2& dcmOffsetNominal, const double& sigmaNominal);
+
+        /**
+         * Set or update the lower and the upper bounds
+         * @param zmpPosition This vector includes the current value of stance foot position ;
+         * @param convexHullVector The convex hull vector related to allowable next step position;
+         * @param stepDuration The nominal value of step timing ;
+         * @param stepDurationTollerance The tollerance of the max and min step timing value with respect to the nominal value ;
+         * @param remainingSingleSupportDuration The remained amount of single support duration ;
+         * @param omega The natural frequency of LIPM ;
+         * @return true/false in case of success/failure.
+         */
+        bool setBoundsVectorOfConstraints(const iDynTree::Vector2& zmpPosition, const iDynTree::VectorDynSize& convexHullVector,
+                                          const double& stepDuration, const double& stepDurationTollerance,
+                                          const double& remainingSingleSupportDuration,const double& omega);
+
+        /**
+         * Get the state of the solver.
+         * @return true if the solver is initialized false otherwise.
+         */
+        bool isInitialized();
+
+        /**
+         * Initialize the solver.
+         * @return true/false in case of success/failure.
+         */
+        bool initialize();
+
+        /**
+         * Solve the optimization problem.
+         * @return true/false in case of success/failure.
+         */
+        bool solve();
+
+        /**
+         * Get the solver solution
+         * @return the entire solution of the solver
+         */
+        iDynTree::VectorDynSize getSolution();
```

Comment:
Here you should return a ``const`` reference to the vector to avoid memory allocation every time you call this method, i.e.
```suggestion
        const iDynTree::VectorDynSize& getSolution() const;
```
The const at the end is just to tell the user "look, this method does not modify anything inside me".

Reply:
Done!

Reply:
You forgot the ``&``.

Reply:
solved thanks!